### PR TITLE
feat: create core app logic

### LINKED ISSUE #2: Create core app logic
Create new class Transaction in logic/transaction folder with following methods:
 * init()
 * check_account_balance(account_id, amount)
 * transfer(sender_account, receiver_account, amount)
    * sender_account is Account instance
    * receiver_account  is Account instance

### PR DESCRIPTION
**Pull request by Oleksandr6676**
it implements core app transaction logic

Closes #2

### CODE REVIEW

**Comment by Oleksandr6676** [comment]
@JanMate Could you please take a look at this implementation, it's a little bit different from the task (has some extra methods)

**Comment by Oleksandr6676** [comment]
also I'm not quite sure what's wrong with linter, built-in linter in pycharm doesn't show any errors

**Comment by JanMate** [comment]
> also I'm not quite sure what's wrong with linter, built-in linter in pycharm doesn't show any errors

black - opinionated python code formatter thinks that the line is too long and should be split into 3 lines.
Your pycharm probably uses only pylint for code analysis.

File: happy_bank_core/logic/transaction.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+class Transaction:
+
+    def __init__(self, customer_id, balance):
+        self.balance = balance
+        self.customer_id = customer_id
+
+    def __eq__(self, other):
+        return self.customer_id == other.customer_id
```
Comment (JanMate) [inline happy_bank_core/logic/transaction.py:8]: Not needed - we don't need to compare 2 transaction objects so far.

File: happy_bank_core/logic/transaction.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+class Transaction:
+
+    def __init__(self, customer_id, balance):
+        self.balance = balance
+        self.customer_id = customer_id
```
Comment (JanMate) [inline happy_bank_core/logic/transaction.py:5]: needless properties

File: happy_bank_core/logic/transaction.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            self.balance += amount
+            print(f'Following amount {amount} has been added to bank account ID: {self.customer_id}')
+            print(f'Current balance: {self.balance}')
+            return self.balance
+        else:
+            print("Amount cannot be 0 or negative number, please try again")
+            exit()
+
+    @staticmethod
+    def transfer(sender_id, receiver_id, amount):
```
Comment (JanMate) [inline happy_bank_core/logic/transaction.py:31]: This method should receive 2 account objects - sender and receiver, and amount of money to transfer between these 2 accounts.

File: happy_bank_core/logic/transaction.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
+class Transaction:
+
+    def __init__(self, customer_id, balance):
+        self.balance = balance
+        self.customer_id = customer_id
+
+    def __eq__(self, other):
+        return self.customer_id == other.customer_id
+
```
Comment (JanMate) [inline happy_bank_core/logic/transaction.py:9]: The 'check_account_balance(...)' method is missing.

**Review by JanMate — CHANGES_REQUESTED** [review]

File: happy_bank_core/logic/account.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+accounts["#1234"] = {"name": "John Doe", "deposit": 1000}
+accounts["#12345"] = {"name": "Johanna Doe", "deposit": 2000}
+
+
+class Account:
+    def __init__(self, customer):
+        """Gets customer's data from accounts dictionary"""
+        self.id = customer
+        self.name = accounts[self.id]["name"]
+        self.deposit = float(accounts[self.id]["deposit"])
```
Comment (JanMate) [inline happy_bank_core/logic/account.py:11]: Account should be such a data class without parsing logic in init method. If we need to parse json, let's create it afterward.

File: happy_bank_core/logic/transaction.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+                "Accounts are equal, please enter correct account ID\n"
+                "----------------------------------------------------"
+            )
+        elif not Transaction.check_balance(sender, amount):
+            print(
+                "insufficient funds or incorrect amount was entered, "
+                "please try again\n"
+                "----------------------------------------------------"
+            )
+        else:
```
Comment (JanMate) [inline happy_bank_core/logic/transaction.py:33]: Could we add a custom exception that's raised when the condition is True? E.g. TransactionException with message property that inherits from general Exception. Then, we can have only 2 if conditions and get rid of else branch that returns a tuple of updated Account instances.

The print commands can be left here - will be replaced by logger call later.

**Review by JanMate — CHANGES_REQUESTED** [review]

File: happy_bank_core/logic/account.py (hunk 5 above)
Comment (Oleksandr6676) [reply happy_bank_core/logic/account.py:11]: So we do not need accounts dictionary at the moment and we are going to pass all arguments manually to Account class, right?

Comment (JanMate) [reply happy_bank_core/logic/account.py:11]: If you mean manually just like this, than yes:
```suggestion
        self.id = id
        self.full_name = full_name
        self.balance = balance
```

**Review by JanMate — APPROVED** [review]
Now it's perfect

**Pull request merged**